test/e2e/util.go

```diff
@@ -695,3 +696,57 @@ func isDualStackCluster(nodes *v1.NodeList) bool {
 	}
 	return false
 }
+
+// used to inject OVN specific test actions
+func wrappedTestFramework(basename string) *framework.Framework{
+	f := framework.NewDefaultFramework(basename)
+	// inject dumping dbs on failure
+	ginkgo.JustAfterEach(func() {
+		if ! ginkgo.CurrentGinkgoTestDescription().Failed {
+			return
+		}
+
+		ovnDocker := "ovn-control-plane"
+
+		logLocation := "/var/log"
+		dbLocation := "/var/lib/openvswitch"
+		ovsdbLocation := "/etc/origin/openvswitch"
+		dbs := []string{"ovnnb_db.db", "ovnsb_db.db"}
+		ovsdb := "conf.db"
+
+		testName :=  strings.Replace(ginkgo.CurrentGinkgoTestDescription().TestText, " ", "_", -1)
+		logDir := fmt.Sprintf("%s/e2e-dbs/%s-%s", logLocation, testName, f.UniqueName)
+
+		var args []string
+
+		// grab all OVS dbs
+		nodes, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		framework.ExpectNoError(err)
+		for _, node := range nodes.Items {
+			// ensure e2e-dbs directory with test case exists
+			args = []string{"docker", "exec", node.Name, "mkdir", "-p", logDir}
+			_, err = runCommand(args...)
+			framework.ExpectNoError(err)
+
+			// node name is the same in kapi and docker
+			args = []string{"docker", "exec", node.Name, "cp", "-f", fmt.Sprintf("%s/%s", ovsdbLocation, ovsdb),
+				fmt.Sprintf("%s/%s", logDir, fmt.Sprintf("%s-%s", node.Name, ovsdb))}
+			_, err = runCommand(args...)
+			framework.ExpectNoError(err)
+		}
+
+		args = []string{"docker", "exec", ovnDocker, "stat", fmt.Sprintf("%s/%s", dbLocation, dbs[0])}
+		_, err = runCommand(args...)
+		framework.ExpectNoError(err)
+
+		// grab the OVN dbs
+		for _, db := range dbs {
```

Comment:
nit: shall we move the ovn db collection to above ovs db collection, this seems more straightforwad at least if the worker nodes are down and we can't grab ovs db's for whatever reason we'll still have ovn dbs, but I'm with this current approach as well, don't want to overthink.

Reply:
since it uses docker commands it means the docker container would have to be gone, i dont think that is likely